odoo/addons/base/res/res_currency.py

```diff
@@ -169,34 +174,38 @@ def is_zero(self, amount):
         return tools.float_is_zero(amount, precision_rounding=self.rounding)
 
     @api.model
-    def _get_conversion_rate(self, from_currency, to_currency):
+    def _get_conversion_rate(self, from_currency, to_currency, company_id, date):
         from_currency = from_currency.with_env(self.env)
         to_currency = to_currency.with_env(self.env)
-        return to_currency.rate / from_currency.rate
+        currency_rates = (from_currency + to_currency).get_rates(company_id, date)
```

Comment:
Should probably use `|` in case someone calls `_get_conversion_rate` directly with the same currency. Or maybe check for that and return `1.0` directly?

Also does this still need the with_env thing?